src/components/StatisticsCard/BattleStyle.js

```diff
@@ -3,7 +3,8 @@ import { OverlayTrigger, Tooltip as Tips } from 'react-bootstrap';
 import LineChart from '../Charts/LineChart';
 
 const BattleStyle = (props) => {
-  const lastSnapshot = props.tankStats.data.snapshots.at(-1);
+  const stats = props.tankStatsCard ? props.tankStatsCard : props.accountStats?.data;
```

Comment:
To simplify the code, you can utilize the logical OR (||) operator.
```suggestion
  const stats = props.tankStatsCard || props.accountStats?.data;
```